## Pandas참조 후 titanic 데이터를 가져오기
- name열이 index로 설정

In [2]:
import pandas as pd
tt=pd.read_excel('titanic3.xls', index_col='name')
tt

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
...                                                 ...       ...     ...   
Zabour, Miss. Hileni                                  3         0  female   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Mapriededer                             3         0    male   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                     age  sibsp  parch  \
name                                                                     
Allen, Miss. Elisabeth Walton                    29.0000      0      0   
Allison, Master. Hudson Trevor                    0.9167      1      2   
Allison, Miss. Helen Loraine                      2.0000      1      2   
Allison, Mr. Hudson Joshua Creighton             30.0000      1      2   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000      1      2   
...                                                  ...    ...    ...   
Zabour, Miss. Hileni                             14.5000      1      0   
Zabour, Miss. Thamine                                NaN      1      0   
Zakarian, Mr. Mapriededer                        26.5000      0      0   
Zakarian, Mr. Ortin                              27.0000      0      0   
Zimmerman, Mr. Leo                               29.0000      0      0   

                                                 ticket      fare    cabin  \
name                                                                         
Allen, Miss. Elisabeth Walton                     24160  211.3375       B5   
Allison, Master. Hudson Trevor                   113781  151.5500  C22 C26   
Allison, Miss. Helen Loraine                     113781  151.5500  C22 C26   
Allison, Mr. Hudson Joshua Creighton             113781  151.5500  C22 C26   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  113781  151.5500  C22 C26   
...                                                 ...       ...      ...   
Zabour, Miss. Hileni                               2665   14.4542      NaN   
Zabour, Miss. Thamine                              2665   14.4542      NaN   
Zakarian, Mr. Mapriededer                          2656    7.2250      NaN   
Zakarian, Mr. Ortin                                2670    7.2250      NaN   
Zimmerman, Mr. Leo                               315082    7.8750      NaN   

                                                embarked boat   body  \
name                                                                   
Allen, Miss. Elisabeth Walton                          S    2    NaN   
Allison, Master. Hudson Trevor                         S   11    NaN   
Allison, Miss. Helen Loraine                           S  NaN    NaN   
Allison, Mr. Hudson Joshua Creighton                   S  NaN  135.0   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)        S  NaN    NaN   
...                                                  ...  ...    ...   
Zabour, Miss. Hileni                                   C  NaN  328.0   
Zabour, Miss. Thamine                                  C  NaN    NaN   
Zakarian, Mr. Mapriededer                              C  NaN  304.0   
Zakarian, Mr. Ortin                                    C  NaN    NaN   
Zimmerman, Mr. Leo                                     S  NaN    NaN   

                                                                       home.dest  
name                                             

## 데이터프레임의 columns만 조회

In [4]:
tt.columns

Index(['pclass', 'survived', 'sex', 'age', 'sibsp', 'parch', 'ticket', 'fare',
       'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')

## .drop()을 이용해서 2개의 열을 삭제
- home.dest
- embarked
- sibsp
- 원본에 반영

In [6]:
tt.drop(columns=['home.dest', 'embarked', 'sibsp'], inplace=True)
tt.head()

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   

                                                     age  parch  ticket  \
name                                                                      
Allen, Miss. Elisabeth Walton                    29.0000      0   24160   
Allison, Master. Hudson Trevor                    0.9167      2  113781   
Allison, Miss. Helen Loraine                      2.0000      2  113781   
Allison, Mr. Hudson Joshua Creighton             30.0000      2  113781   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000      2  113781   

                                                     fare    cabin boat   body  
name                                                                            
Allen, Miss. Elisabeth Walton                    211.3375       B5    2    NaN  
Allison, Master. Hudson Trevor                   151.5500  C22 C26   11    NaN  
Allison, Miss. Helen Loraine                     151.5500  C22 C26  NaN    NaN  
Allison, Mr. Hudson Joshua Creighton             151.5500  C22 C26  NaN  135.0  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  151.5500  C22 C26  NaN    NaN

## del을 이용한 열 삭제
- parch

In [8]:
del tt['parch']
tt.head()

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   

                                                     age  ticket      fare  \
name                                                                         
Allen, Miss. Elisabeth Walton                    29.0000   24160  211.3375   
Allison, Master. Hudson Trevor                    0.9167  113781  151.5500   
Allison, Miss. Helen Loraine                      2.0000  113781  151.5500   
Allison, Mr. Hudson Joshua Creighton             30.0000  113781  151.5500   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000  113781  151.5500   

                                                   cabin boat   body  
name                                                                  
Allen, Miss. Elisabeth Walton                         B5    2    NaN  
Allison, Master. Hudson Trevor                   C22 C26   11    NaN  
Allison, Miss. Helen Loraine                     C22 C26  NaN    NaN  
Allison, Mr. Hudson Joshua Creighton             C22 C26  NaN  135.0  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  C22 C26  NaN    NaN

## 사망자 명단 조회하기

In [10]:
dead_mask=tt['survived']==0
tt[dead_mask]

pclass  survived     sex  \
name                                                                        
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mr. Hudson Joshua Creighton                  1         0    male   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
Andrews, Mr. Thomas Jr                                1         0    male   
Artagaveytia, Mr. Ramon                               1         0    male   
...                                                 ...       ...     ...   
Zabour, Miss. Hileni                                  3         0  female   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Mapriededer                             3         0    male   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                  age    ticket      fare  \
name                                                                        
Allison, Miss. Helen Loraine                      2.0    113781  151.5500   
Allison, Mr. Hudson Joshua Creighton             30.0    113781  151.5500   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0    113781  151.5500   
Andrews, Mr. Thomas Jr                           39.0    112050    0.0000   
Artagaveytia, Mr. Ramon                          71.0  PC 17609   49.5042   
...                                               ...       ...       ...   
Zabour, Miss. Hileni                             14.5      2665   14.4542   
Zabour, Miss. Thamine                             NaN      2665   14.4542   
Zakarian, Mr. Mapriededer                        26.5      2656    7.2250   
Zakarian, Mr. Ortin                              27.0      2670    7.2250   
Zimmerman, Mr. Leo                               29.0    315082    7.8750   

                                                   cabin boat   body  
name                                                                  
Allison, Miss. Helen Loraine                     C22 C26  NaN    NaN  
Allison, Mr. Hudson Joshua Creighton             C22 C26  NaN  135.0  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  C22 C26  NaN    NaN  
Andrews, Mr. Thomas Jr                               A36  NaN    NaN  
Artagaveytia, Mr. Ramon                              NaN  NaN   22.0  
...                                                  ...  ...    ...  
Zabour, Miss. Hileni                                 NaN  NaN  328.0  
Zabour, Miss. Thamine                                NaN  NaN    NaN  
Zakarian, Mr. Mapriededer                            NaN  NaN  304.0  
Zakarian, Mr. Ortin                                  NaN  NaN    NaN  
Zimmerman, Mr. Leo                                   NaN  NaN    NaN  

[809 rows x 9 columns]

## 시신 식별번호(body)가 없는 데이터 조회하기

In [12]:
tt.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1309 entries, Allen, Miss. Elisabeth Walton to Zimmerman, Mr. Leo
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   pclass    1309 non-null   int64  
 1   survived  1309 non-null   int64  
 2   sex       1309 non-null   object 
 3   age       1046 non-null   float64
 4   ticket    1309 non-null   object 
 5   fare      1308 non-null   float64
 6   cabin     295 non-null    object 
 7   boat      486 non-null    object 
 8   body      121 non-null    float64
dtypes: float64(3), int64(2), object(4)
memory usage: 102.3+ KB


In [14]:
tt['body'].isna().sum()

1188

In [16]:
unkown_body_mask=tt['body'].isna()
tt[unkown_body_mask]

pclass  survived     sex  \
name                                                                        
Allen, Miss. Elisabeth Walton                         1         1  female   
Allison, Master. Hudson Trevor                        1         1    male   
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
Anderson, Mr. Harry                                   1         1    male   
...                                                 ...       ...     ...   
Yousif, Mr. Wazli                                     3         0    male   
Yousseff, Mr. Gerious                                 3         0    male   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                     age  ticket      fare  \
name                                                                         
Allen, Miss. Elisabeth Walton                    29.0000   24160  211.3375   
Allison, Master. Hudson Trevor                    0.9167  113781  151.5500   
Allison, Miss. Helen Loraine                      2.0000  113781  151.5500   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0000  113781  151.5500   
Anderson, Mr. Harry                              48.0000   19952   26.5500   
...                                                  ...     ...       ...   
Yousif, Mr. Wazli                                    NaN    2647    7.2250   
Yousseff, Mr. Gerious                                NaN    2627   14.4583   
Zabour, Miss. Thamine                                NaN    2665   14.4542   
Zakarian, Mr. Ortin                              27.0000    2670    7.2250   
Zimmerman, Mr. Leo                               29.0000  315082    7.8750   

                                                   cabin boat  body  
name                                                                 
Allen, Miss. Elisabeth Walton                         B5    2   NaN  
Allison, Master. Hudson Trevor                   C22 C26   11   NaN  
Allison, Miss. Helen Loraine                     C22 C26  NaN   NaN  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  C22 C26  NaN   NaN  
Anderson, Mr. Harry                                  E12    3   NaN  
...                                                  ...  ...   ...  
Yousif, Mr. Wazli                                    NaN  NaN   NaN  
Yousseff, Mr. Gerious                                NaN  NaN   NaN  
Zabour, Miss. Thamine                                NaN  NaN   NaN  
Zakarian, Mr. Ortin                                  NaN  NaN   NaN  
Zimmerman, Mr. Leo                                   NaN  NaN   NaN  

[1188 rows x 9 columns]

살아있는 사람은 시신 식별번호가 없는 것은 당연

## 사망자이나 시신식별번호가 없는 명단 찾기

In [18]:
tt[dead_mask & unkown_body_mask]

pclass  survived     sex  \
name                                                                        
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
Andrews, Mr. Thomas Jr                                1         0    male   
Baumann, Mr. John D                                   1         0    male   
Baxter, Mr. Quigg Edmond                              1         0    male   
...                                                 ...       ...     ...   
Yousif, Mr. Wazli                                     3         0    male   
Yousseff, Mr. Gerious                                 3         0    male   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                  age    ticket      fare  \
name                                                                        
Allison, Miss. Helen Loraine                      2.0    113781  151.5500   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0    113781  151.5500   
Andrews, Mr. Thomas Jr                           39.0    112050    0.0000   
Baumann, Mr. John D                               NaN  PC 17318   25.9250   
Baxter, Mr. Quigg Edmond                         24.0  PC 17558  247.5208   
...                                               ...       ...       ...   
Yousif, Mr. Wazli                                 NaN      2647    7.2250   
Yousseff, Mr. Gerious                             NaN      2627   14.4583   
Zabour, Miss. Thamine                             NaN      2665   14.4542   
Zakarian, Mr. Ortin                              27.0      2670    7.2250   
Zimmerman, Mr. Leo                               29.0    315082    7.8750   

                                                   cabin boat  body  
name                                                                 
Allison, Miss. Helen Loraine                     C22 C26  NaN   NaN  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  C22 C26  NaN   NaN  
Andrews, Mr. Thomas Jr                               A36  NaN   NaN  
Baumann, Mr. John D                                  NaN  NaN   NaN  
Baxter, Mr. Quigg Edmond                         B58 B60  NaN   NaN  
...                                                  ...  ...   ...  
Yousif, Mr. Wazli                                    NaN  NaN   NaN  
Yousseff, Mr. Gerious                                NaN  NaN   NaN  
Zabour, Miss. Thamine                                NaN  NaN   NaN  
Zakarian, Mr. Ortin                                  NaN  NaN   NaN  
Zimmerman, Mr. Leo                                   NaN  NaN   NaN  

[688 rows x 9 columns]

사망자가 809명인데, 식별번호가 없는 명단이 688명이나 된다.

## 사망자이나 시신식별번호가 없는 명단으로 새로운 df 만들기
- `notfound_list`

In [20]:
notfound_list=tt[dead_mask & unkown_body_mask]
notfound_list

pclass  survived     sex  \
name                                                                        
Allison, Miss. Helen Loraine                          1         0  female   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)       1         0  female   
Andrews, Mr. Thomas Jr                                1         0    male   
Baumann, Mr. John D                                   1         0    male   
Baxter, Mr. Quigg Edmond                              1         0    male   
...                                                 ...       ...     ...   
Yousif, Mr. Wazli                                     3         0    male   
Yousseff, Mr. Gerious                                 3         0    male   
Zabour, Miss. Thamine                                 3         0  female   
Zakarian, Mr. Ortin                                   3         0    male   
Zimmerman, Mr. Leo                                    3         0    male   

                                                  age    ticket      fare  \
name                                                                        
Allison, Miss. Helen Loraine                      2.0    113781  151.5500   
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  25.0    113781  151.5500   
Andrews, Mr. Thomas Jr                           39.0    112050    0.0000   
Baumann, Mr. John D                               NaN  PC 17318   25.9250   
Baxter, Mr. Quigg Edmond                         24.0  PC 17558  247.5208   
...                                               ...       ...       ...   
Yousif, Mr. Wazli                                 NaN      2647    7.2250   
Yousseff, Mr. Gerious                             NaN      2627   14.4583   
Zabour, Miss. Thamine                             NaN      2665   14.4542   
Zakarian, Mr. Ortin                              27.0      2670    7.2250   
Zimmerman, Mr. Leo                               29.0    315082    7.8750   

                                                   cabin boat  body  
name                                                                 
Allison, Miss. Helen Loraine                     C22 C26  NaN   NaN  
Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  C22 C26  NaN   NaN  
Andrews, Mr. Thomas Jr                               A36  NaN   NaN  
Baumann, Mr. John D                                  NaN  NaN   NaN  
Baxter, Mr. Quigg Edmond                         B58 B60  NaN   NaN  
...                                                  ...  ...   ...  
Yousif, Mr. Wazli                                    NaN  NaN   NaN  
Yousseff, Mr. Gerious                                NaN  NaN   NaN  
Zabour, Miss. Thamine                                NaN  NaN   NaN  
Zakarian, Mr. Ortin                                  NaN  NaN   NaN  
Zimmerman, Mr. Leo                                   NaN  NaN   NaN  

[688 rows x 9 columns]

## notfound_list의 승객 등급별 개수 확인

In [24]:
notfound_list['pclass'].value_counts()
#잘못된 분석! 

pclass
3    473
2    127
1     88
Name: count, dtype: int64

3등급(하류층)이 많으나, 이건 당연할 수 있다. 비율을 구해보자.

## 전체 titanic 전체 승객 등급별 개수 확인

In [26]:
tt['pclass'].value_counts()

pclass
3    709
1    323
2    277
Name: count, dtype: int64

## 승객 등급별 실종 비율 확인

In [57]:
(notfound_list['pclass'].value_counts() / tt['pclass'].value_counts())*100

pclass
1    27.244582
2    45.848375
3    66.713681
Name: count, dtype: float64

## 1. 사망자의 하류층 티겟 가격 분포와 2. 생존자의 하류층 티켓 가격 분포를 확인해보자.

In [32]:
survived_mask=tt['survived']==1
pclass_3_mask=tt['pclass']==3


In [34]:
dead_pclass3_mask=tt[dead_mask & pclass_3_mask]
#dead_pclass3_mask['fare'].value_counts(normalize=True)

In [36]:
survived_pclass3_mask=tt[survived_mask & pclass_3_mask]
#survived_pclass3_mask.value_counts(subset='fare',normalize=True)

In [40]:
dead_pclass3_mask.describe()

pclass  survived         age        fare        body
count   528.0     528.0  370.000000  527.000000   55.000000
mean      3.0       0.0   25.974549   13.435451  155.818182
std       0.0       0.0   11.880771   12.033986  102.403720
min       3.0       0.0    0.333300    0.000000    1.000000
25%       3.0       0.0   19.000000    7.750000   67.500000
50%       3.0       0.0   25.000000    8.050000  153.000000
75%       3.0       0.0   32.875000   14.500000  257.000000
max       3.0       0.0   74.000000   69.550000  328.000000

In [42]:
survived_pclass3_mask.describe()

pclass  survived         age        fare  body
count   181.0     181.0  131.000000  181.000000   0.0
mean      3.0       1.0   21.545166   12.916919   NaN
std       0.0       0.0   11.607635    9.775015   NaN
min       3.0       1.0    0.166700    0.000000   NaN
25%       3.0       1.0   16.000000    7.750000   NaN
50%       3.0       1.0   22.000000    8.050000   NaN
75%       3.0       1.0   29.000000   15.741700   NaN
max       3.0       1.0   63.000000   56.495800   NaN

3등급에서 티켓가격의 차이가 생존유무와 관련성이 있어보이지 않는다.